### PR TITLE
CAMEL-13185: Add splitResults implementations to Olingo Components

### PR DESCRIPTION
**Pull request by phantomjinx**
* Adds splitResults methods to both Olingo 2 & 4 Consumers

### CODE REVIEW

File: components/camel-olingo4/camel-olingo4-component/src/test/java/org/apache/camel/component/olingo4/Olingo4ComponentTest.java [hunk 1]
```diff
@@ -402,6 +409,49 @@ public void testProducerReadFilterAlreadySeen() throws Exception {
...
+    public void testConsumerReadSplitResults() throws Exception {
+        final Map<String, Object> headers = new HashMap<>();
+        String endpoint = "olingo4://read/People?consumer.splitResult=true";
+        int expectedEntities = 20;
+        final ClientEntitySet entities = (ClientEntitySet)requestBodyAndHeaders(endpoint, null, headers);
+        assertNotNull(entities);
+        assertEquals(expectedEntities, entities.getEntities().size());
+
+        int expectedMsgCount = 3;
+        MockEndpoint mockEndpoint = getMockEndpoint("mock:consumer-splitresult");
```
Comment (davsclaus) [inline components/camel-olingo4/camel-olingo4-component/src/test/java/org/apache/camel/component/olingo4/Olingo4ComponentTest.java]: You should set your mock expectations before you send data into Camel, so move these lines up in the beginning of the test method, then send data, and then assert on the mock. Do the same in the other unit tests.

File: components/camel-olingo4/camel-olingo4-component/src/test/java/org/apache/camel/component/olingo4/Olingo4ComponentTest.java [hunk 2]
```diff
@@ -402,6 +409,49 @@ public void testProducerReadFilterAlreadySeen() throws Exception {
...
+        // each containing a different entity.
+        //
+        for (int i = 0; i < expectedMsgCount; ++i) {
+            Object body = mockEndpoint.getExchanges().get(i).getIn().getBody();
+            assertTrue(body instanceof ClientEntity);
+            ClientEntity entity = (ClientEntity)body;
+            ClientProperty nameProperty = entity.getProperty("UserName");
+            assertNotNull(nameProperty);
+
+            switch(i) {
```
Comment (davsclaus) [inline components/camel-olingo4/camel-olingo4-component/src/test/java/org/apache/camel/component/olingo4/Olingo4ComponentTest.java]: There is likely a checkstyle issue here, run mvn install -P checkstyle to see its errors.

**Comment by davsclaus** [comment]
It would be good to add a section to the documentation (the adoc file in the src/main/docs folder) about this option to use splitResults, and provide a little example.

**Comment by davsclaus** [comment]
@phantomjinx are you able to work on the review suggestions?

**Comment by phantomjinx** [comment]
Yeah. Was going to look at them this afternoon. If you would like to update the docs I won't be offended ;-)?

File: components/camel-olingo4/camel-olingo4-component/src/test/java/org/apache/camel/component/olingo4/Olingo4ComponentTest.java (hunk 1 above)
Comment (phantomjinx) [reply components/camel-olingo4/camel-olingo4-component/src/test/java/org/apache/camel/component/olingo4/Olingo4ComponentTest.java]: Done.

File: components/camel-olingo4/camel-olingo4-component/src/test/java/org/apache/camel/component/olingo4/Olingo4ComponentTest.java (hunk 2 above)
Comment (phantomjinx) [reply components/camel-olingo4/camel-olingo4-component/src/test/java/org/apache/camel/component/olingo4/Olingo4ComponentTest.java]: Done.

**Comment by phantomjinx** [comment]
@davsclaus

Actually, looking at the adoc files, splitResult has already been described:
- https://github.com/apache/camel/blob/master/components/camel-olingo4/camel-olingo4-component/src/main/docs/olingo4-component.adoc#consumer-endpoints
- https://github.com/apache/camel/blob/master/components/camel-olingo2/camel-olingo2-component/src/main/docs/olingo2-component.adoc#consumer-endpoints

Not sure whether an example will add to that, WDYT?

cc @oscerd

**Comment by oscerd** [comment]
I think it's enough for the docs for the moment.

**Comment by phantomjinx** [comment]
@davsclaus @oscerd 
If you guys are ok with the resolved changes to this PR, are you good to merge it?

**Comment by oscerd** [comment]
Yes, I'm going to merged this

**Comment by oscerd** [comment]
Merged on master. @phantomjinx can you please do the same for the camel-2.x branch? Thanks!

**Pull request closed without merging**